### ISSUE TITLE
Announce results of Autocomplete component

### ISSUE THREAD

**Issue by marcoscv-work** [liferay/clay#5613]
Announce results of Autocomplete component with LiveAnnouncer

After testing this with VoiceOver and JAWS, it looks like the number of results is not being announced, but I can see the changes in the previous PR: https://github.com/liferay/clay/pull/5593

So probably something missing.

clayui.com > autocomplete
<img width="546" alt="image" src="https://github.com/liferay/clay/assets/7963804/15c5f9e0-6e5c-41d7-999a-6a32895f53a9">

storybook.clayui.com > autocomplete
<img width="550" alt="image" src="https://github.com/liferay/clay/assets/7963804/d8454a70-1418-4728-b37b-7349613678bd">

**Comment by matuzalemsteles** [comment]
@marcoscv-work sorry about that actually the clayui.com site is not containing the recent changes of this update, you can test this at next.clayui.com, we will be sending the updates at clayui.com and storybook.clayui.com today with the release.

See this with Voice Over testing at next.clayui.com:
![Screenshot 2023-07-03 at 13 35 17](https://github.com/liferay/clay/assets/13750819/768c7a73-eb35-4299-88f5-f9a2b447c3da)

Closing this for now, if this still persists we can reopen anyway.
